Library/RSBot.Core/Objects/Cos/Cos.cs

```diff
@@ -234,6 +234,72 @@ public void MoveTo(Position destination)
             Thread.Sleep(Convert.ToInt32((distance / Game.Player.ActualSpeed) * 10000));
         }
 
+        /// <summary>
+        /// Purchases the item.
+        /// </summary>
+        /// <param name="tab">The tab.</param>
+        /// <param name="slot">The slot.</param>
+        /// <param name="amount">The amount.</param>
+        public bool PurchaseItemFromNpc(int tab, int slot, ushort amount)
+        {
+            var npc = Game.SelectedEntity;
+            if (npc == null)
+            {
+                Log.Debug("Cannot buy items for cos, because no shop is selected!");
+                return false;
+            }
+
+            var packet = new Packet(0x7034);
+            packet.WriteByte(0x13); //Buy item flag
```

Comment:
It would be nice to have a enumeration for all the different inventory operations:

```
public enum InventoryOperation : byte
{
    SP_UPDATE_SLOTS_INV = 0,
    SP_UPDATE_SLOTS_CHEST = 1,
    SP_DEPOSIT_ITEM = 2,
    SP_WITHDRAW_ITEM = 3,
    SP_SET_XCHANGE_ITEM = 4, //SP_ADD_EXCHANGE
    SP_CANCEL_XCHANGE_ITEM = 5, //SP_DEL_EXCHANGE
    SP_PICK_ITEM = 6,
    SP_DROP_ITEM = 7,
    SP_BUY_ITEM = 8,
    SP_SELL_ITEM = 9,
    SP_DROP_GOLD = 10,
    SP_DEPOSIT_GOLD = 11,
    SP_WITHDRAW_GOLD = 12,
    SP_SET_XCHANGE_GOLD = 13, //SP_UPDATE_EXCHANGE_GOLD
    
    SP_ADD_ITEM_BY_SERVER = 14,
    SP_DEL_ITEM_BY_SERVER = 15,

    SP_UPDATE_SLOTS_INV_COS = 16,
    SP_PICK_ITEM_COS = 17,
    SP_DROP_ITEM_COS = 18,
    SP_BUY_ITEM_COS = 19,
    SP_SELL_ITEM_COS = 20,
    SP_ADD_COSITEM_BY_SERVER = 21,
    SP_DEL_COSITEM_BY_SERVER = 22,

    //Missing 23
    SP_BUY_CASH_ITEM = 24,

    //Missing 25
    SP_MOVE_ITEM_PET_PC = 26,

    SP_MOVE_ITEM_PC_PET = 27,
    SP_PICK_ITEM_BY_OTHER = 28,
    SP_GUILD_CHEST_UPDATE_SLOT = 29,
    SP_GUILD_CHEST_DEPOSIT_ITEM = 30,
    SP_GUILD_CHEST_WITHDRAW_ITEM = 31,
    SP_GUILD_CHEST_DEPOSIT_GOLD = 32,
    SP_GUILD_CHEST_WITHDRAW_GOLD = 33,
    SP_RESTORE_SOLDITEM_INSHOP = 34,
    SP_MOVE_ITEM_AVATAR_PC = 35,
    SP_MOVE_ITEM_PC_AVATAR = 36,

    //Missing 37
    SP_MOVE_ITEM_TRADE_NOW = 38,

    SP_PUSH_ITEM_INTO_MAGIC_CUBE = 39,
    SP_POP_ITEM_FROM_MAGIC_CUBE = 40,
    SP_DEL_ITEM_IN_MAGIC_CUBE = 41,
    SP_ACTIVATE_MAGIC_CUBE = 42,
    SP_BUY_ITEM_WITH_TOKEN = 43,
    SP_PICK_SPECIAL_ITEM = 44,

    //Missing 45-51

    SP_PICK_SPECIAL_ITEM_BY_SILKPET = 52,
    //Missing 53
    SP_PICK_SPECIAL_ITEM_BY_OTHER = 54,
}
```

Reply:
done